types_test.go

```diff
@@ -64,3 +65,60 @@ func TestUnmarshalMerkleTreeLeaf(t *testing.T) {
 		}
 	}
 }
+
+func mustB64Decode(s string) []byte {
+	b, err := base64.StdEncoding.DecodeString(s)
+	if err != nil {
+		panic(err)
+	}
+	return b
+}
+
+func TestToSignedTreeHead(t *testing.T) {
+	validRootHash := mustB64Decode("cImB6R0Uh8Kp6pAatajQU8E0hYWvzbXhB79gwMHSD8A=")
+	validSignature := mustB64Decode("BAMARzBFAiAH+1rjzqjwdrU0oBqaGeYGJcbMcHBMbBp8iLMNj2fUrwIhAIQNN7jy+c4TTnTu/aagwq0DTVkbeFzcSXPExPXQPwQ5")
+
+	tests := []struct {
+		desc      string
+		rootHash  []byte
+		signature []byte
+		wantErr   bool
```

Comment:
Might be nice to be able to check more error detail -- could have `wantErr string` and if it's non-empty do a `strings.Contains` check.   (cf. `TestLogEntryFromLeaf`)

Reply:
There is general advice to not do string checking as part of checking errors, to avoid test fragility in the case where error strings may be reworded.  While only checking a few words from the error messages makes it less at risk to this, I tend to avoid it unless absolutely necessary to differentiate between error types - in which case I'd be more likely to return different error types and check the type rather than check the string in any way.

Reply:
<bikeshed>
I think it depends on who's emitting the errors -- if they're from some base library and getting passed on, then yes it's rather brittle, but if the code under test is creating the error text then I think it's fine to check strings to confirm all the different error arms are hit.
</bikeshed>
Fine as-is.

Reply:
(Hah, GitHub completely ate my mock (unescaped) bikeshed delimiters!)